### PR TITLE
Edit wse draft

### PR DESCRIPTION
**Pull request by uwlnf**

### CODE REVIEW

File: Backend/Controllers/AuthenticationController.cs [hunk 1]
```diff
@@ -113,8 +113,8 @@ public IQueryable<Session> GetSessions([FromAuthentication] User user)
 
     [RequiresAuthentication]
     [HttpPost("change-password-uwu")]
-    public async Task PasswordChange([FromAuthentication] User user, string newPassword) {
-        ChangePassword(user, newPassword);
+    public async Task PasswordChange([FromAuthentication] User user, ChangePasswordDto newPassword) {
+        ChangePassword(user, newPassword.NewPassword);
```
Comment (Saalvage) [inline Backend/Controllers/AuthenticationController.cs:117]: 1. Should not send the duplicate password for no reason
2. Requiring the correct old password should harden security, so we should do it

File: Backend/Controllers/SearchController.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
 using Models;
+using Models.DTO;
```
Comment (Saalvage) [inline Backend/Controllers/SearchController.cs:8]: Redundant

File: Frontend/Pages/Components/AnswerComponent.razor [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        <div class="col">
+            <h6 class="fw-bold mt-1 mb-1">
+                @AnswerCreator.FirstNames @AnswerCreator.LastName
+                <small class="text-muted">@Answer.CreationTime</small>
+            </h6>
+            @Answer.Text
+        </div>
+    </div>
+    @if (Answer.CreatorId == _userId)
+    {
```
Comment (Saalvage) [inline Frontend/Pages/Components/AnswerComponent.razor:19]: Wrong codestyle

File: Frontend/Pages/Components/AnswerComponent.razor [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    AnswerId="AnswerIdToDelete"
+    OnClose="@OnDeleteAnswerDialogClose"/>
+}
+
+@code {
+    [Inject]
+    private HttpClient HttpClient { get; set; } = default!;
+
+    [Parameter]
+    public Question Question { get; set; } = null!;
```
Comment (Saalvage) [inline Frontend/Pages/Components/AnswerComponent.razor:44]: Inconsistent, either we're going with `default!` or with `null!`. Idc which one, but not both.

File: Frontend/Pages/Components/CollabModal.razor [hunk 5]
```diff
@@ -8,51 +12,83 @@
...
-                <label class="form-label mt-2">E-Mail oder Name des neuen Kollaborators</label>
-                <div class="ui-field">
-                    <input class="form-control"/>
-                </div>
-                <div class="d-grid gap-2 mt-3 mb-2 d-flex justify-content-end">
-                    <button type="button" class="btn btn-success" @onclick="@ModalAddCollab">Hinzufügen</button>
-                </div>
+                }
+                
+                @if (_isLoggedIn && Array.Exists(Collabs, c => c.UserId == UserId))
```
Comment (Saalvage) [inline Frontend/Pages/Components/CollabModal.razor:41]: Wrong style again

File: Models/Question.cs [hunk 6]
```diff
@@ -2,9 +2,10 @@
 using System.ComponentModel.DataAnnotations.Schema;
 using System.Text.Json.Serialization;
 
-namespace Models; 
+namespace Models;
 
-public class Question {
+public class Question
+{
```
Comment (Saalvage) [inline Models/Question.cs:8]: No

File: Models/WseRights.cs [hunk 7]
```diff
@@ -1,7 +1,8 @@
 ﻿namespace Models;
 
 [Flags]
-public enum WseRights {
+public enum WseRights
+{
```
Comment (Saalvage) [inline Models/WseRights.cs:5]: Grrr

**Review by Saalvage — CHANGES_REQUESTED** [review]
Mainly reviewed backend and models, but the frontend sure looks like a mess!!

**Review by Saalvage — APPROVED** [review]
Backend & Model changes approved, largely left the Frontend alone, up to the devs working on it to decide whether the codestyle should be fixed.

**Comment by JohHub** [comment]
not fully functional but merging anyway because branch got way out of hand

**Pull request merged**